dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,408 @@
+    private static final String MOD_USER = "mod_user";
+    private static final String SORT_ORDER = "sort_order";
+    private static final String IDENTIFIER = "identifier";
+    private static final String SHOW_ON_MENU = "show_on_menu";
+    private static final String FRIENDLY_NAME = "friendly_name";
+    private static final String DRAWED = "drawed";
+    private static final String MAX_CONTENTLETS = "max_contentlets";
+    private static final String USE_DIV = "use_div";
+    private static final String STATICIFY = "staticify";
+    private static final String ADD_CONTAINER_LINKS = "add_container_links";
+    private static final String CONTAINERS_ADDED = "containers_added";
+    private static final String INODE = "inode";
+
+    private ConvertToPOJOUtil(){
+
+    }
+
+    /**
+     * Creates new instances of T given a List<Map> of DB results
+     * @param results
+     * @param classToUse
+     * @param <T>
+     * @return
+     * @throws Exception
+     */
+    public static<T> List<T> convertDotConnectMapToPOJO(List<Map<String,String>> results, final Class classToUse)
+            throws ParseException, IllegalAccessException, InvocationTargetException, InstantiationException, NoSuchMethodException, NoSuchFieldException {
+
+        List<T> ret = null;
+
+        if(results == null || results.isEmpty()){
+            return Collections.emptyList();
+        }
+
+        if (Folder.class.equals(classToUse)){
+            ret = (List<T>) convertDotConnectMapToFolder(results);
+        }
+
+        if (Container.class.equals(classToUse)){
+            ret = (List<T>) convertDotConnectMapToContainer(results);
+        }
+
+        if (Link.class.equals(classToUse)){
+            ret = (List<T>) convertDotConnectMapToLink(results);
+        }
+
+        if (Identifier.class.equals(classToUse)){
+            ret = (List<T>) convertDotConnectMapToIdentifier(results);
+        }
+
+        if (Template.class.equals(classToUse)){
+            ret = (List<T>) convertDotConnectMapToTemplate(results);
+        }
+
+        if (ret != null){
+            return ret;
+        }
+
+        return getMapFields(results, classToUse);
+    }
+
+    @NotNull
+    private static <T> List<T> getMapFields(List<Map<String, String>> results, Class classToUse)
+            throws NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException, NoSuchFieldException, ParseException {
+        List<T> ret;
+        Map<String, String> properties;
+        ret = new ArrayList<>();
+
+        for (final Map<String, String> map : results) {
+            Constructor<?> ctor = classToUse.getConstructor();
+            final T object = (T) ctor.newInstance();
+
+            properties = map.entrySet().stream().collect(Collectors
+                    .toMap(entry -> CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, entry.getKey()), entry ->map.get(entry.getKey())));
+
+            for (final String property: properties.keySet()){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)